app/services/order_service.py

```diff
@@ -0,0 +1,182 @@
+from uuid import UUID
+from datetime import datetime, timezone
+from decimal import Decimal
+from fastapi import HTTPException, status
+from sqlalchemy.orm import Session,joinedload
+from app.models import Order,Product,Status,OrderProduct
+from app.schemas import CreateOrderRequestModel, CreateOrderResponseModel, GetOrderResponseModel, OrderProductBaseModel, UpdateOrderStatusResponseModel
+
+
+class OrderService:
+    def __init__(self, db: Session):
+        self.db = db
+
+    def create_order(self, user_id: UUID, order_request: CreateOrderRequestModel) -> CreateOrderResponseModel:
+        # Step 1: Get pending status
+        pending_status = self._get_pending_status()
+
+        # Step 2: Fetch and validate products
+        product_map, total_price = self._validate_products(order_request.products)
+
+        # Step 3: Create new order
+        new_order = self._create_new_order(user_id, pending_status.id, total_price)
+
+        # Step 4: Create order products and update stock
+        self._create_order_products(new_order.id, order_request.products, product_map)
+
+        # Step 5: Create response model
+        response_data = CreateOrderResponseModel(
+            id=new_order.id,
+            user_id=new_order.user_id,
+            status=pending_status.name,
+            total_price=new_order.total_price,
+            created_at=new_order.created_at
+        )
+
+        return response_data
+
+    def _get_pending_status(self):
```

Comment:
It's not ideal to keep fetching statuses for each order. Instead, we should load all the statuses once into memory (query all statuses and store them in a dictionary with the status name as the key). Don't worry if you don't fully understand this or know how to implement it right now—you can keep it as is for this phase.